xdsl/dialects/ptr.py

```diff
@@ -91,6 +91,16 @@ class ToPtrOp(IRDLOperation):
     assembly_format = "$source attr-dict `:` type($source) `->` type($res)"
 
 
+@irdl_op_definition
+class FromPtrOp(IRDLOperation):
+    name = "ptr_xdsl.from_ptr"
+
+    source = operand_def(PtrType)
+    res = result_def(MemRefType)
+
+    assembly_format = "$source attr-dict `:` type($source) `->` type($res)"
```

Comment:
The `type($source)` isn't necessary but I guess it makes the intent clearer.

Also, `functional-type` would put brackets round the operand types, I'm not sure if this is preferred by mlir, or if there is precedent for doing it this way too.

Reply:
I followed the syntax in MLIR, which has some more potential information on the ptr type itself, such as memory space, which we haven't added yet.

Reply:
Ah, it wasn't clear to me that this was based off an mlir op (especially as the dialect is called `ptr_xdsl`)

Reply:
It's called that because they haven't merged their things yet, and we're living in the future :)